br/pkg/restore/snap_client/client.go

```diff
@@ -1000,13 +1000,14 @@ func (rc *SnapClient) execChecksum(
 		zap.String("table", tbl.OldTable.Info.Name.O),
 	)
 
-	if tbl.OldTable.NoChecksum() {
-		logger.Warn("table has no checksum, skipping checksum")
+	expectedChecksumStats := metautil.CalculateChecksumStatsOnFiles(tbl.OldTable.Files)
+	if !expectedChecksumStats.ChecksumExists() {
```

Comment:
not sure if we need to do the check here, if an empty table is being backed up, should we check after restore it's still empty?

Reply:
Yes, the error log is misleading.